rust_snuba/src/processors/utils.rs

```diff
@@ -39,3 +40,38 @@ where
     };
     Ok(seconds_since_epoch.timestamp() as u32)
 }
+
+#[allow(dead_code)]
+pub struct CogsAccountant {
+    accountant: UsageAccountant<KafkaProducer>,
+}
+
+impl CogsAccountant {
+    #[allow(dead_code)]
+    fn new(bootstrap_servers: &str) -> Self {
+        let config = KafkaConfig::new_producer_config(bootstrap_servers, None);
+        Self {
+            accountant: UsageAccountant::new_with_kafka(config, None, None),
+        }
+    }
+
+    fn record_bytes(&mut self, resource_id: &str, app_feature: &str, amount_bytes: u64) {
+        if let Err(err) =
+            self.accountant
+                .record(resource_id, app_feature, amount_bytes, UsageUnit::Bytes)
+        {
+            tracing::warn!(?err, "error recording cogs");
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_record_cogs() {
+        let mut accountant = CogsAccountant::new("127.0.0.1:9092");
+        accountant.record_bytes("generic_metrics_processor_sets", "ddm", 100)
```

Comment:
fyi, the app_feature for ddm is `custom`

Reply:
ok may as well update to reflect reality